packages/astro/src/compiler/index.ts

```diff
@@ -115,6 +115,10 @@ export async function compileComponent(source: string, { compileOptions, filenam
 import fetch from 'node-fetch';
 ${result.imports.join('\n')}
 
+if(!('fetch' in globalThis)) {
```

Comment:
👌 im not one to leave code reviews, just like this bit of logic, first time seeing logic like this. (they dont have that as an example on MDN😅 )